File: clients/backend/psychopass/communitypsychoPasses.js
class CommunityPsychoPasses {
    url;
    constructor(url = `${process.env.BACKEND_URL}/psychopass/community`) {
        this.url = url;
    }

    async get(communityID) {
        try {
            const response = await fetch(`${this.url}?id=${communityID}`, {
                method: "GET",
                headers: {
                    "Content-type": "application/json",
                    "User-Agent": `sibyl-discord/${process.env.npm_package_version} node.js/${process.version}`,
                    "Authorization": `Token ${process.env.BACKEND_API_KEY}`
                }
            });
            if (!response.ok) return response.status
            return response.json();
        } catch (error) {
            console.error(error);
        }
    }

    async create(communityID) {
        try {
            const response = await fetch(this.url, {
                method: "POST",
                headers: {
                    "Content-type": "application/json",
                    "User-Agent": `sibyl-discord/${process.env.npm_package_version} node.js/${process.version}`,
                    "Authorization": `Token ${process.env.BACKEND_API_KEY}`
                },
                body: JSON.stringify({ communityID })
            });
            if (!response.ok) return response.status
        } catch (error) {
            console.error(error);
        }
    }

    async delete(communityID) {
        try {
            const response = await fetch(`${this.url}?id=${communityID}`, {
                method: "DELETE",
                headers: {
                    "Content-type": "application/json",
                    "User-Agent": `sibyl-discord/${process.env.npm_package_version} node.js/${process.version}`,
                    "Authorization": `Token ${process.env.BACKEND_API_KEY}`
                }
            });
            if (!response.ok) return response.status
        } catch (error) {
            console.error(error);
        }
    }

}

const communityPsychoPasses = new CommunityPsychoPasses();
export default communityPsychoPasses;

File: clients/backend/psycho_pass/community_psycho_passes.py
from os import getenv
from sys import version
from requests import get, put


class CommmunityPsychoPasses:
    url = f"{getenv('BACKEND_URL')}/psychopass/community"

    @classmethod
    def read(cls, community_id: int) -> dict:
        response = get(
            f"{cls.url}?id={community_id}",
            headers={
                "Accept": "application/json",
                "User-Agent": f"sibyl-discord python/{version}",
                "Authorization": f"Token {getenv('BACKEND_API_KEY')}",
            },
        )

        response.raise_for_status()
        return response.json()

    # For Removing a User Psycho-Pass from a Community Psycho-Pass Only
    @classmethod
    def update(cls, community_id: int, user_id: int) -> dict:
        response = put(
            cls.url,
            headers={
                "Accept": "application/json",
                "User-Agent": f"sibyl-discord python/{version}",
                "Authorization": f"Token {getenv('BACKEND_API_KEY')}",
            },
            json={"communityID": community_id, "userID": user_id},
        )

        response.raise_for_status()
        return response.json()
